src/storage/access_observer.cpp

```diff
@@ -0,0 +1,21 @@
+#include "storage/access_observer.h"
+#include "storage/block_compactor.h"
+
+namespace terrier::storage {
+void AccessObserver::ObserveGCInvocation() {
+  gc_epoch_++;
+  for (auto it = last_touched_.begin(), end = last_touched_.end(); it != end;) {
+    if (it->second + COLD_DATA_EPOCH_THRESHOLD < gc_epoch_) {
+      compactor_->PutInQueue(it->first);
+      it = last_touched_.erase(it);
```

Comment:
Should we be asserting that erase succeeded?

Reply:
No, because this is calling the erase method on a valid iterator handed out by the unordered_map, which will always succeed. In fact I don't think this version of erase even  bothers with error behavior.